homeassistant/components/esphome/bluetooth/client.py

```diff
@@ -56,38 +57,41 @@ def mac_to_int(address: str) -> int:
     return int(address.replace(":", ""), 16)
 
 
+def _on_disconnected(task: asyncio.Task[Any], fut: asyncio.Future[None]) -> None:
+    if task and not task.done():
+        task.cancel()
+
+
 def verify_connected(func: _WrapFuncType) -> _WrapFuncType:
     """Define a wrapper throw BleakError if not connected."""
 
     async def _async_wrap_bluetooth_connected_operation(
         self: ESPHomeClient, *args: Any, **kwargs: Any
     ) -> Any:
-        loop = self._loop  # pylint: disable=protected-access
-        disconnected_futures = (
-            self._disconnected_futures  # pylint: disable=protected-access
-        )
+        # pylint: disable=protected-access
+        loop = self._loop
+        disconnected_futures = self._disconnected_futures
         disconnected_future = loop.create_future()
         disconnected_futures.add(disconnected_future)
-
         task = asyncio.current_task(loop)
-
-        def _on_disconnected(fut: asyncio.Future[None]) -> None:
-            if task and not task.done():
-                task.cancel()
-
-        disconnected_future.add_done_callback(_on_disconnected)
+        disconnect_handler = partial(_on_disconnected, task)
```

Comment:
https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.add_done_callback this is the recommended way to pass the argument